pkg/devfile/devfile.go

```diff
@@ -899,8 +901,25 @@ func ValidateDevfile(log logr.Logger, URL string) (shouldIgnoreDevfile bool, dev
 
 	devfileData, err := ParseDevfile(devfileSrc)
 	if err != nil {
-		log.Error(err, fmt.Sprintf("failed to parse the devfile content from %s", URL))
-		return shouldIgnoreDevfile, nil, fmt.Errorf(fmt.Sprintf("err: %v, failed to parse the devfile content from %s", err, URL))
+		var newErr error
+		if merr, ok := err.(*multierror.Error); ok {
+			for i := range merr.Errors {
+				switch merr.Errors[i].(type) {
+				case *devfileValidation.MissingDefaultCmdWarning:
+					log.Info(fmt.Sprintf("devfile is missing default command, found a warning: %v", merr.Errors[i]))
+				default:
+					newErr = multierror.Append(newErr, merr.Errors[i])
+				}
+			}
+		} else {
+			newErr = err
+		}
+		if newErr != nil {
+			if merr, ok := newErr.(*multierror.Error); !ok || len(merr.Errors) != 0 {
+				log.Error(newErr, fmt.Sprintf("failed to parse the devfile content from %s", URL))
+				return shouldIgnoreDevfile, nil, fmt.Errorf(fmt.Sprintf("err: %v, failed to parse the devfile content from %s", newErr, URL))
+			}
+		}
```

Comment:
```suggestion
		if merr, ok := err.(*multierror.Error); ok {
			for i := range merr.Errors {
				switch merr.Errors[i].(type) {
				case *devfileValidation.MissingDefaultCmdWarning:
					log.Info(fmt.Sprintf("devfile is missing default command, found a warning: %v", merr.Errors[i]))
				default:
					newErr = multierror.Append(newErr, merr.Errors[i])
				}
			}
		} else if !ok || len(merr.Errors) != 0 {
				log.Error(newErr, fmt.Sprintf("failed to parse the devfile content from %s", URL))
				return shouldIgnoreDevfile, nil, fmt.Errorf(fmt.Sprintf("err: %v, failed to parse the devfile content from %s", newErr, URL))
		}
```

looks like we can simplify it like this?

Reply:
if not a multierror, newErr needs to be assigned to `err`